snugget_load.py

```diff
@@ -0,0 +1,205 @@
+    (snuggetID, row["text"], row["heading"], row["image"], row["intensity"])
+  )
+
+
+
+def readColumnsFrom(appName, table, cur):
+  cols = []
+  cur.execute(
+    "SELECT column_name FROM information_schema.columns WHERE table_name = %s;", 
+    [appName + "_" + table.lower()]
+  )
+  for row in cur.fetchall():
+    cols.append(row[0])
+  return cols
+
+
+
+
+
+def getSectionID(appName, sectionName, cur):
+  cur.execute("SELECT MIN(id) FROM " + appName + "_snuggetsection WHERE name = %s;", [sectionName])
+  sectionID = cur.fetchone()[0]
+  if sectionID is not None:
+    return sectionID
+  else: # if no sectionID was found then we need to create the section
+    cur.execute("INSERT INTO " + appName + "_snuggetsection(name) VALUES(%s);", [sectionName])
+    cur.execute("SELECT id FROM " + appName + "_snuggetsection WHERE name = %s;", [sectionName])
+    sectionID = cur.fetchone()[0]
+    return sectionID
+  
+  
+
+
+def findFilterID(appName, shapefile, key, cur):
+  cols = readColumnsFrom(appName, shapefile, cur)
+  keyColumn = cols[1]
+  cur.execute("SELECT id FROM " + appName + "_" + shapefile + " WHERE " + keyColumn + "::text = %s;", [key])
+  return str(cur.fetchone()[0])
+
+    
+    
+def findAllFilterIDs(appName, shapefile, cur):
+  ids = []
+  cur.execute("SELECT id FROM " + appName + "_" + shapefile + ";")
+  for row in cur.fetchall():
+    ids.extend(row)
+  return ids
+
+
+
+def getSnuggetID(appName, sectionID, filterColumn, filterID, cur):
+  cur.execute(
+    'SELECT MAX(id) FROM ' + appName + '_snugget WHERE section_id = %s AND "' + filterColumn + '" = %s;',
+    (sectionID, filterID)
+  )
+  snuggetID = cur.fetchone()[0]
+  if snuggetID is not None:
+    return str(snuggetID)
+  else:
+    return None
+
+
+
+
+def checkForSnugget(appName, sectionID, filterColumn, filterID, cur):
+  try:
+    snuggetID = getSnuggetID(appName, sectionID, filterColumn, filterID, cur)
+    cur.execute("SELECT content FROM " + appName + "_textsnugget WHERE snugget_ptr_id = %s;", [snuggetID])
+    return cur.fetchone()[0]
+  except: # if nothing came back from the DB, just return None rather than failing
+    return None
+
+  
+  
+def removeOldSnugget(appName, sectionID, filterColumn, filterID, cur):
+  snuggetID = getSnuggetID(appName, sectionID, filterColumn, filterID, cur)
+  if snuggetID is not None:
+    cur.execute("DELETE FROM " + appName + "_textsnugget WHERE snugget_ptr_id = %s;", [snuggetID])
+    cur.execute("DELETE FROM " + appName + "_snugget WHERE id = %s;", [snuggetID])
+    print("Replacing snugget", snuggetID)
+    
+
+  
+
+
+# Check with the user about overwriting existing snuggets, giving them the options to:
+# either quit and check what's going on, or say "yes to all" and not get prompted again.
+def askUserAboutOverwriting(row, oldSnugget, oldSnuggets, snuggetFile, overwriteAll):
+  if overwriteAll: # if it's already set, then don't do anything else
```

Comment:
It's not in global scope so I'm not changing the structure, but I am renaming it to reduce potential confusion.